src/main/java/org/sonatype/nexus/repository/chef/internal/metadata/util/TgzParser.java

```diff
@@ -18,11 +18,13 @@ public class TgzParser {
 
     @Nullable
     public InputStream getFileFromInputStream(final InputStream is, String fileName) throws IOException {
+        // Match the sought filename only if it is in the root of the archive or one subfolder down
+        String fileNameRegex = "^(?:[^/]*/)?" + fileName.replaceAll("\\.", "\\\\.") + "$";
```

Comment:
Good catch the replace '.' to '\.'; however in theory other regex params can impact you.
Would be then best to use `\Q<string>\E`, or the java method to do it: `Pattern.quote(fileName)`.

I think in term of code it is shorter to do: `"^(?:[^/]*/)?\\Q" + fileName" + "\\E$";`